app/services/UserServiceImpl.js

```diff
@@ -29,7 +29,28 @@ module.exports = function(userDao) {
     });
   };
 
+  const findByUsername = function(username, password) {
+    return new Promise((resolve, reject) => {
+      _userDao.findByUsername(username)
+        .then((user) => {
+          return isMatchingPassword(password, user.password)
```

Comment:
Took your suggestion and changed the function to resolve with a user should the isMatchingPassword function resolve with true. I'm unsure whether or not the two .catch lines are needed but I included them just incase.

`const authenticateUser = function(emailOrUsername, password) {
    return new Promise((resolve, reject) => {
      _userDao.getByEmailOrUsername(emailOrUsername)
        .then((user) => {
          return isMatchingPassword(password, user.password)
            .then((isMatching) => {
              isMatching ? resolve(user) : reject();
            })
            .catch(reject);
        })
        .catch(reject);
    });
  };`